server/schedule/operator/builder.go

```diff
@@ -163,17 +213,19 @@ func (b *Builder) SetPeers(peers map[uint64]*metapb.Peer) *Builder {
 	if b.err != nil {
 		return b
 	}
-	b.targetPeers = peersMap{}
-	for k, p := range peers {
-		if p.GetStoreId() != k {
-			b.err = errors.Errorf("setPeers with mismatch storeID: %v", peers)
+
+	for key, peer := range peers {
+		if peer == nil || key == 0 || peer.GetStoreId() != key || core.IsInJointState(peer) {
+			b.err = errors.Errorf("setPeers with mismatch peers: %v", peers)
```

Comment:
If the error is caused by `IsInJointState`, can we distinguish it from the error message here?

Reply:
Yes, the `peer.Role` in the `peers` of the log can be observed.